Source/bmalloc/libpas/src/libpas/pas_allocation_mode.h

```diff
@@ -31,13 +31,19 @@
 PAS_BEGIN_EXTERN_C;
 
 enum pas_allocation_mode {
-    /* We are allocating an object and intend to store its address as a compact
-       pointer. */
-    pas_compact_allocation_mode,
-
-    /* We are allocating an ordinary object and will not store its address in
-       any compact pointer type. */
+    /* We are allocating an object from ordinary memory and don't plan on
+       compacting its address. */
     pas_non_compact_allocation_mode,
+
+    /* We are allocating an object from ordinary memory and expect to
+       be able to compact its address, but don't expect all addresses in
+       that memory to be trivially compactible. */
+    pas_maybe_compact_allocation_mode,
+
+    /* We are allocating an object from memory where all addresses within
+       that memory are expected to be trivially compactible, like in the
+       immortal heap. */
```

Comment:
Is it a requirement that all addresses must be trivially compactible? As the comment is written, it sounds like the caller may need to handle the case where an address isn't compactible, in which case it's hard to understand the difference between this and the "maybe" case from the comments.

Reply:
Yeah I agree, I'll drop the "expected to be" so it's clear all memory in the range _is_ trivially compactible, without exceptions.